### PR TITLE
feat: Add deploy option

### PR DESCRIPTION
**Pull request by maekawataiki**
*Issue #, if available:*

*Description of changes:*
- Small fix to improve deploy experience (unique Cognito domain name, suppress alb log nag error)
- Add deploy option features which user can enable/disable easily
  - Security Related
    - Self-Signup
    - Email domain restriction
    - WAF (IP restriction)
    - SAML Authentication
    - Custom Domain with SSL
  - Cost Related
    - Spot Instance
    - Scale down automatically
    - Scheduled scale up / down
- Documentation Update

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by HouseoLogy** [comment]
Hi @maekawataiki 
great extensions overall. It does improve the setup in terms of user experience, potentially cost and security as-well. 

Could you please just update the default settings like commented. Then I'll merge it asap.

Many thanks

File: docs/DEPLOY_OPTION.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+```json
+  "samlAuthEnabled": true,
+```
+- samlAuthEnabled: Setting this to `true` will switch to the SAML-only authentication screen. The conventional authentication functionality using Cognito user pools will no longer be available.
+
+## Cost-related Settings
+
+### Spot Instance
+
+You can use spot instance to reduce cost for non-critical workload. Set `useSpot` to `true` in the context. You may also modify `spotPrice`. Instance will be available only when spot price is below `spotPrice`.
```
Comment (HouseoLogy) [inline docs/DEPLOY_OPTION.md:99]: Default is set to true

Comment (maekawataiki) [reply docs/DEPLOY_OPTION.md]: Thank you. Fixed it.

File: docs/DEPLOY_OPTION.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+cdk deploy
+```
+
+## Security Related Settings
+
+### Disable Self Sign-Up
+
+You may disable user self-signup so administrator can fully control user from Cognito Console. 
+
+Set `selfSignUpEnabled` to `false` in the context. (The default is `true`)
```
[the remark this answers is not in the document]
Comment (maekawataiki) [reply docs/DEPLOY_OPTION.md]: Thank you. Fixed it.

**Pull request merged**